### DISCUSSION TITLE
Unexpected restart

### DISCUSSION THREAD

**Discussion by 9thChasingWindGirl** [AdguardTeam/AdGuardHome#6852] [Q&A] [answered]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, ARM64

### Installation

Custom package (OpenWrt, HomeAssistant, etc; please mention in the description)

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

v0.107.46

### Action

When openwrt handles downstream traffic, if the peak traffic is too large, it will cause adguardhome to restart unexpectedly, during which all other network connections will be interrupted except the traffic downloaded by bt. In another case, when using clash, there is a high probability of this error when switching network nodes frequently.
[AdGuardHome3-21-20_49.log](https://github.com/AdguardTeam/AdGuardHome/files/14695414/AdGuardHome3-21-20_49.log)

### Expected result

Methods to reduce or avoid this error.

### Actual result

none

### Additional information and/or screenshots

![屏幕截图 2024-03-21 205628](https://github.com/AdguardTeam/AdGuardHome/assets/69580996/9508357d-62b9-49e6-b068-22ff4b435da2)

**Comment by Cebeerre** [comment]
Hi, 

Seems that you're using a heavily modified kind of fork of OpenWRT bundled with a lot of other stuff to bypass DPIs and the likes. I'd suggest that you open an issue in their repo: [https://github.com/kiddin9/OpenWrt_x86-r2s-r4s-r5s-N1]([https://github.com/kiddin9/OpenWrt_x86-r2s-r4s-r5s-N1](https://github.com/AdguardTeam/AdGuardHome/issues/url))

**Comment by 9thChasingWindGirl** [comment]
> Hi, 
> 
> Seems that you're using a heavily modified kind of fork of OpenWRT bundled with a lot of other stuff to bypass DPIs and the likes. I'd suggest that you open an issue in their repo: [https://github.com/kiddin9/OpenWrt_x86-r2s-r4s-r5s-N1](url)
> 

I'm a little confused. What do you mean by DPIs

**Comment by Cebeerre** [comment]
Deep Pakcket Inspection [https://en.wikipedia.org/wiki/Deep_packet_inspection](url)

**Comment by 9thChasingWindGirl** [comment]
> Deep Pakcket Inspection [https://en.wikipedia.org/wiki/Deep_packet_inspection](url)

The author of the opewnrt branch I'm using is unlikely to handle issues. Is there anything I can do to improve the situation on the basis of what I have?

**Comment by Cebeerre** [comment]
> > Deep Pakcket Inspection [https://en.wikipedia.org/wiki/Deep_packet_inspection](url)
> 
> The author of the opewnrt branch I'm using is unlikely to handle issues. Is there anything I can do to improve the situation on the basis of what I have?

It's just a guess, but probably the restarts are actually happening due to events fired in ubus and then managed by procd. Check the hotplug scripts (inside ```/etc/hotplug.d```) and see if there's anything that might be triggering these restarts .

Hope that helps !

**Comment by 9thChasingWindGirl** [comment]
> > > Deep Pakcket Inspection [https://en.wikipedia.org/wiki/Deep_packet_inspection](url)
> > 
> > The author of the opewnrt branch I'm using is unlikely to handle issues. Is there anything I can do to improve the situation on the basis of what I have?
> 
> It's just a guess, but probably the restarts are actually happening due to events fired in ubus and then managed by procd. Check the hotplug scripts (inside ```/etc/hotplug.d```) and see if there's anything that might be triggering these restarts .
> 
> Hope that helps !
> 
> 

Forgive me for not understanding how openwrt works, but the question I submitted, I guess, has to do with the number of filtering rules I use in my router. I use adguard as the server for dns requests, smartdns is the source of dns.![屏幕截图 2024-03-25 201413.png](https://github.com/AdguardTeam/AdGuardHome/assets/69580996/b02119b6-0ff1-4a46-bbe7-f1495035bdcc)

![屏幕截图 2024-03-25 201403.png](https://github.com/AdguardTeam/AdGuardHome/assets/69580996/4eae2466-c8cc-44f6-abfb-a5bfce9bf669)

**Comment by Cebeerre** [comment]
How much RAM does this AX1800 have ?

**Comment by 9thChasingWindGirl** [comment]
> How much RAM does this AX1800 have ?

512 MB

**Comment by Cebeerre** [comment]
I'd suggest then checking how's the router dealing with your amount of rules and the available memory:

![image](https://github.com/AdguardTeam/AdGuardHome/assets/44902718/041c2af2-cba4-49f0-96a5-9b5179826da5)

I've something around 1,3 milion rules, and AGH takes roughly between 250-300 Mb of RAM, maybe your router is too tight on memory to expect loading so much rules ...

**Reply by 9thChasingWindGirl** [reply]
This is not a serious problem, thank you very much for your guidance and answers👍

**Comment by Cebeerre** [comment, marked as answer]
Let me move this to a discussion, as this does not seem to be related to a bug.

**Reply by 9thChasingWindGirl** [reply]
Hello ceberre,I'm sorry to bother you again, but I modified the partition table in openwrt and used the 512MB size for swap. Does this help? I also need to add a point. When this problem occurs, adh will show that it is not running but redirected. I googled this problem and found that it may be related to the number of repeated DNS queries. All in all, I have not completely solved this problem, but I am trying.

**Reply by Cebeerre** [reply]
What I'd do:

- Try stopping any unnecessary services that you might be running in the router.
- Try lowering the cache memory in AGH itself
- Try setting up smaller lists like OISD Small [https://oisd.nl/setup](https://oisd.nl/setup)

Keep in mind that AGH is quite memory hungry ... it's not like making the dnsmasq native to load some blocklists, so these kind of routers are far from being a perfect fit. Something with 1Gb would do a better job ...

**Reply by 9thChasingWindGirl** [reply]
Yes, I've taken some steps, but I'm still skeptical that expanding the swap partition will mitigate the error of not running adh? So far, there have been no errors in adh, except that dns requests seem to be slowing down.

**Reply by Cebeerre** [reply]
Well, it might work ... not sure where you placed your swap file (an external USB ?) but keep in mind that this might impact wear sensitive flash devices and it will probably slow down the whole thing :) 

You can play as well with the swapiness and see how it goes: 

```
# sysctl -w vm.swappiness=
```

Giving a value between 0 and 100, being 0 rare RAM swap out and 100 very agressive RAM swap out ... to make it persistent to reboots, place your setting in ```/etc/sysctl.d```

**Reply by 9thChasingWindGirl** [reply]
This is the original partition table of my equipment AX1800.
`Device            Start       End   Sectors  Size Type

/dev/mmcblk0p1       34      1569      1536  768K unknown
/dev/mmcblk0p2     1570      2081       512  256K unknown
/dev/mmcblk0p3     2082      2593       512  256K unknown
/dev/mmcblk0p4     2594      6177      3584  1.8M unknown
/dev/mmcblk0p5     6178      9761      3584  1.8M unknown
/dev/mmcblk0p6     9762     10273       512  256K unknown
/dev/mmcblk0p7    10274     10785       512  256K unknown
/dev/mmcblk0p8    10786     11297       512  256K unknown
/dev/mmcblk0p9    11298     11809       512  256K unknown
/dev/mmcblk0p10   11810     12321       512  256K unknown
/dev/mmcblk0p11   12322     12833       512  256K unknown
/dev/mmcblk0p12   12834     13345       512  256K unknown
/dev/mmcblk0p13   13346     14625      1280  640K unknown
/dev/mmcblk0p14   14626     15905      1280  640K unknown
/dev/mmcblk0p15   15906     16417       512  256K unknown
/dev/mmcblk0p16   16418     28705     12288    6M unknown
/dev/mmcblk0p17   28706     40993     12288    6M unknown
/dev/mmcblk0p18   40994    163873    122880   60M unknown
/dev/mmcblk0p19  163874    172065      8192    4M unknown
/dev/mmcblk0p20  172066    294945    122880   60M unknown
/dev/mmcblk0p21  294946    303137      8192    4M unknown
/dev/mmcblk0p22  303138    344097     40960   20M unknown
/dev/mmcblk0p23  344098    345121      1024  512K unknown
/dev/mmcblk0p24  345122    524321    179200 87.5M unknown
/dev/mmcblk0p25  524322   1138721    614400  300M unknown
/dev/mmcblk0p26 1138722   2187297   1048576  512M unknown
/dev/mmcblk0p27 2187298 119783390 117596093 56.1G unknown`

mmcblk0p27 exists as a file system in the original system,when I found that there are still two partitions in the partition table, mmcblk0p25 and 26 have relatively large space to use, I used 25 to expand overlay and 26 as swap partition.

This is my current partition table.

`Device            Start       End   Sectors  Size Type

/dev/mmcblk0p1       34      1569      1536  768K unknown
……
/dev/mmcblk0p18   40994    163873    122880   60M unknown
/dev/mmcblk0p19  163874   1138721    974848  476M Linux filesystem
/dev/mmcblk0p20 1138722   2187297   1048576  512M Linux filesystem
/dev/mmcblk0p21 2187298 119783390 117596093 56.1G Linux filesystem`

Mmcblk0p1, 2 ... 17 and 18 partitions have not been modified. I rudely merged 19-25 into one partition to expand the overlay.



![屏幕截图 2024-04-03 163846](https://github.com/AdguardTeam/AdGuardHome/assets/69580996/b6c320a7-81bd-4d7b-9c27-8369983d7ebb)
